src/Storages/System/StorageSystemDDLWorkerQueue.cpp

```diff
@@ -85,8 +85,18 @@ static String clusterNameFromDDLQuery(ContextPtr context, const DDLTask & task)
 
     String description = fmt::format("from {}", task.entry_path);
     ParserQuery parser_query(end, settings[Setting::allow_settings_after_format_in_insert]);
-    ASTPtr query = parseQuery(
-        parser_query, begin, end, description, settings[Setting::max_query_size], settings[Setting::max_parser_depth], settings[Setting::max_parser_backtracks]);
+    ASTPtr query;
+
+    try
+    {
+        query = parseQuery(
+            parser_query, begin, end, description, settings[Setting::max_query_size], settings[Setting::max_parser_depth], settings[Setting::max_parser_backtracks]);
+    }
+    catch (Exception &)
+    {
+        /// Best effort - ignore parse error and present available information
+        return "UNKNOWN";
```

Comment:
This still may overlaps with existing cluster, let's maybe convert `cluster` to `Nullable`, and:
- use `NULL` when it was not `ON CLUSTER` query (is it possible?)
- empty value during parsing error
And we may want to log this exception (so that the smart user can see it in logs)

And also describe this in a column comment

Reply:
I don't think that it is possible to have queries without ON CLUSTER here.
Probably NULL is slightly better than empty string in case of parse error: we use NULL if we cannot e.g. identify host.
Agree?